### DISCUSSION TITLE
Why adding different ipynb files?

### DISCUSSION THREAD

**Discussion by sanjay-kv** [recodehive/Stackoverflow-Analysis#87] [Q&A] [answered]
@Saksh8 I see different ipynb files updating on the folders. 
Ideally we need only one file.

https://github.com/Recode-Hive/Stackoverflow-Analysis/commits/main/Stackoverflow_Survey_Analysis(1).ipynb

**Comment by Saksh8** [comment]
Sir when I was removing the old file from your project a branch conflict was constantly arising there while making a pull request.
This was there even when I uploaded the same file with the same name. In order to resolve the conflict I had to keep the old one also.

You can remove the old one from your side. I don't have access to do that. both ipynb files are same except the heat maps that are being added to the [](https://github.com/Recode-Hive/Stackoverflow-Analysis/commits/main/Stackoverflow_Survey_Analysis(1).ipynb)

**Reply by Hassassin-10** [reply, marked as answer]
Organization: Helps keep your code clean and organized by separating different tasks, experiments, or sections of your project into their own notebooks.

Modularity: Allows you to focus on specific parts of your project individually without cluttering one large notebook with too much information.

Reusability: Encourages reuse of code and functions across different notebooks without duplicating code, making it easier to manage and update.

Collaboration: Facilitates collaboration by allowing team members to work on different parts of the project simultaneously without conflicts.

Documentation: Helps in documenting different steps or processes in your project more clearly and systematically.

**Comment by HackxAnkit** [comment]
> @Saksh8 I see different ipynb files updating on the folders. Ideally we need only one file.
> 
> https://github.com/Recode-Hive/Stackoverflow-Analysis/commits/main/Stackoverflow_Survey_Analysis(1).ipynb

If multiple people are working on the project simultaneously or if you have multiple sessions open, each session might be creating or updating its own notebook file

**Reply by Hassassin-10** [reply]
Organization: Helps keep your code clean and organized by separating different tasks, experiments, or sections of your project into their own notebooks.

Modularity: Allows you to focus on specific parts of your project individually without cluttering one large notebook with too much information.

Reusability: Encourages reuse of code and functions across different notebooks without duplicating code, making it easier to manage and update.

Collaboration: Facilitates collaboration by allowing team members to work on different parts of the project simultaneously without conflicts.

Documentation: Helps in documenting different steps or processes in your project more clearly and systematically.

**Comment by RaviSingh733** [comment]
Adding multiple .ipynb (Jupyter Notebook) files to a project can serve various purposes, such as:

Modularity and Organization: Splitting work across different notebooks allows you to keep related sections of code and analyses separate. For instance, you might have separate notebooks for data cleaning, exploratory data analysis (EDA), and

**Reply by Hassassin-10** [reply]
Organization: Helps keep your code clean and organized by separating different tasks, experiments, or sections of your project into their own notebooks.

Modularity: Allows you to focus on specific parts of your project individually without cluttering one large notebook with too much information.

Reusability: Encourages reuse of code and functions across different notebooks without duplicating code, making it easier to manage and update.

Collaboration: Facilitates collaboration by allowing team members to work on different parts of the project simultaneously without conflicts.

Documentation: Helps in documenting different steps or processes in your project more clearly and systematically.

**Comment by Hassassin-10** [comment]
Organization: Helps keep your code clean and organized by separating different tasks, experiments, or sections of your project into their own notebooks.

Modularity: Allows you to focus on specific parts of your project individually without cluttering one large notebook with too much information.

Reusability: Encourages reuse of code and functions across different notebooks without duplicating code, making it easier to manage and update.

Collaboration: Facilitates collaboration by allowing team members to work on different parts of the project simultaneously without conflicts.

Documentation: Helps in documenting different steps or processes in your project more clearly and systematically.

**Comment by FlavioNinin** [comment]
It sounds like you're dealing with multiple ".ipynb"  (Jupyter Notebook) files being updated across folders, when ideally you'd prefer a single consolidated notebook!
